Code/Framework/AzFramework/Platform/Linux/AzFramework/Input/Buses/Notifications/RawInputNotificationsBus_xcb.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/EBus/EBus.h>
+
+#include <xcb/xcb.h>
+
+namespace AzFramework
+{
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    //! EBus interface used to listen for raw xcb input as broadcast by the system. Applications
+    //! that want xcb events to be processed by the AzFramework input system must broadcast all
+    //! input events received by the native input handle, which is the lowest level we can get input.
+    //!
+    //! It's possible to receive multiple events per button/key per frame, and it is likely that
+    //! xcb input events will not be dispatched from the main thread, so care should be taken to
+    //! ensure thread safety when implementing event handlers that connect to this xcb event bus.
+    //!
+    //! This EBus is intended primarily for the AzFramework input system to process raw input events.
+    //! Most systems that need to process input should use the generic AzFramework input interfaces,
+    //! but if necessary it is perfectly valid to connect directly to this EBus for xcb events.
+    class RawInputNotificationsXcb : public AZ::EBusTraits
```

Comment:
Generic events can be more than just input notifications. And at this point, I don't think we need to call them 'raw' events either. Can we rename to 'XcbEventHandler'?

Reply:
I'm following the established pattern used in the rest of AzFramework:

```
Code/Framework/AzFramework/Platform/Android/AzFramework/Input/Buses/Notifications/RawInputNotificationBus_Android.h:        virtual void OnRawInputEvent(const AInputEvent* /*rawInputEvent*/) {}
Code/Framework/AzFramework/Platform/Mac/AzFramework/Input/Buses/Notifications/RawInputNotificationBus_Mac.h:        virtual void OnRawInputEvent(const NSEvent* /*nsEvent*/) = 0;
Code/Framework/AzFramework/Platform/Linux/AzFramework/Input/Buses/Notifications/RawInputNotificationsBus_xcb.h:        virtual void OnRawInputEvent(xcb_generic_event_t* event) = 0;
Code/Framework/AzFramework/Platform/Windows/AzFramework/Input/Buses/Notifications/RawInputNotificationBus_Windows.h:        virtual void OnRawInputEvent(const RAWINPUT& /*rawInput*/) {}
```

Does it matter for this to be consistent with the rest of the platforms?

Reply:
Could go either way on this, ultimately I think it comes down to whether we want to do the check for "what type of event is this" at the point where we make the broadcast, or here. On Windows we essentially do the former, and check in the WndpProc function for 'WM_INPUT' events, which are then casted and forwarded over the RawInputNotificationBus_Windows.h

It's tricky here though because there doesn't appear to be any base/generic 'xcb_input_event_t', rather just the base 'xcb_generic_event_t' which encompasses things other than input, which is why I initially leaned towards keeping the  LinuxXcbEventHandler from #3975 and simply broadcasting all received 'xcb_generic_event_t' events over it from both EditorQtApplication::nativeEventFilter and ApplicationLinux_xcb::PumpSystemEventLoopOnce / ApplicationLinux_xcb::PumpSystemEventLoopUntilEmpty (the latter of which is going to give you the events you need to get raw input while running the launcher).

The one downside to keeping the generic LinuxXcbEventHandler from #3975 is that we'll end up broadcasting more events than strictly necessary, such that input devices will receive window related (and other) xcb events that they'll ignore, and the NatvieWindow will receive input related (and other) xcb events that they'll ignore. I'm not sure whether this will make any practical difference vs checking the type of xcb event at the point it's received and broadcasting over different EBuses for different systems (eg. RawInputNotificationBus_Linux + a different one specifically for window related events).

I don't mind terribly which way we decide to go, but the key point is that whatever we do, it should be the same in both EditorQtApplication::nativeEventFilter and ApplicationLinux_xcb::PumpSystemEventLoopOnce / ApplicationLinux_xcb::PumpSystemEventLoopUntilEmpty so that we're delivering 'raw' xcb events in the same fashion for both the editor and launcher.

Reply:
I went ahead and removed this EBus, we'll use the `LinuxXcbEventHandler` instead.